Modules/mathmodule.c

```diff
@@ -1136,11 +1141,10 @@ math_ceil(PyObject *module, PyObject *number)
         }
         if (PyErr_Occurred())
             return NULL;
+        x = PyFloat_AsDouble(number);
+        if (x == -1.0 && PyErr_Occurred())
```

Comment:
This will be partially covered by tests.

I think you could add a test with:
```python
class TestBadCeil:
    def __ceil__(self):
        return "spam"
```

Reply:
After merging #102523 this should be fixed.